### DISCUSSION TITLE
Increase the assertion count when checking for expections

### DISCUSSION THREAD

**Discussion by tflori** [laravel/framework#45172] [Ideas]
At the moment when your test only expects a notification to be sent or an event to be fired or a job to be dispatched it will fail because no assertions where made.

In my opinion that is wrong. Something was asserted.

It is not even that hard we just need to increase the assertion count inside the closure when the expected result happened:
https://github.com/laravel/browser-kit-testing/blob/6.x/src/Concerns/MocksApplicationServices.php#L406
```php
if (($notified === $notifiable ||
     $notified->getKey() == $notifiable->getKey()) &&
    get_class($dispatched['instance']) === $notification
) {
    $this->addToAssertionCount(1);
    return $this;
}
```

**Comment by henzeb** [comment]
There are tools to assert if a notification or jobs are dispatched. You can look them up in the documentation. So I am not sure what you are looking for?

**Comment by tflori** [comment]
I'm not sure that we mean the same. We are using laravel/browser-kit-testing to test calls to routes. We have tests where we use `$this->expectsNotification($user, Notification::class)` - that is an assertion. But with phpunit 8 these tests fail because of "no assertions where made". I'm not looking for a solution how to test that but my idea is that the library should mark the test as positive including the assertion count.

**Comment by crynobone** [comment]
Browser-Kit-Testing is no longer in active development and wouldn't likely receive new feature other than support on upcoming major Laravel releases. 

It might be best to migrate to latest TestCase available from the Framework and use the next Notification assertions.
